AVADA.pm

```diff
@@ -0,0 +1,243 @@
+https://pubmed.ncbi.nlm.nih.gov/31467448/
+
+NB: The plugin currently does not annotate for downstream_gene_variant and upstream_gene_variant.
+
+Pre-requisites
+1) AVADA data is available for GRCh37 and can be downloaded from: 
+http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+
+wget http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+
+2) The file needs to be tabix indexed. You can do this by following commands:
+
+gzip -d avada_v1.00_2016.vcf.gz
+bgzip avada_v1.00_2016.vcf 
+tabix avada_v1.00_2016.vcf.gz
+
+3) As you have already noticed, tabix utility must be installed in your path to use this plugin. 
+
+Options are passed to the plugin as key=value pairs:
+
+The plugin can then be run to retrieve AVADA annotations. 
+By default, the variants are matched with the HGNC gene symbol
+./vep -i variations.vcf --plugin AVADA,file=path/to/file
+
+The output always includes the following column:
+AVADA_GENE_ID: Gene ID associated with variant as reported by AVADA
+
+One of the following columns is included depending on the option passed:
+AVADA_PMID: PubMed ID evidence for the variant as reported by AVADA
+AVADA_PMID_WITH_VARIANT: PubMed ID evidence for the variant as reported by AVADA along with the original variant string
+AVADA_PMID_WITH_REFSEQ: PubMed ID evidence for the variant as reported by AVADA along with RefSeq id
+AVADA_PMID_WITH_REFSEQ_AND_VARIANT: PubMed ID evidence for the variant as reported by AVADA along with RefSeq id and original variant string
+
+The plugin can optionally be run by specifying the feature to match with.
+
+In order to match by HGNC gene symbol:
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=gene_symbol 
+
+In order to match by Ensembl gene identifier :
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=ensembl_gene_id
+
+The plugin can also be run to report the original variant string reported in the publication. 
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,original_variant_string=1
+
+Additionally, the plugin can be run to match RefSeq identifier from VEP cache/database with AVADA data.
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,match_by_refseq=1
+
+=cut
+
+package AVADA;
+
+use strict;
+use warnings;
+use List::MoreUtils qw(uniq);
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub get_header_info {
+  my $self = shift;
+  my %header;
+
+  $header{"AVADA_GENE_ID"} = "Gene ID associated with variant as reported by AVADA" ; 
+  $header{"AVADA_PMID"} = "PubMed ID evidence for the variant as reported by AVADA" if not $self->{original_variant_string} and not $self->{match_by_refseq} ;  
+  $header{"AVADA_PMID_WITH_VARIANT"} = "PubMed ID evidence for the variant as reported by AVADA along with original variant string" if $self->{original_variant_string} and not $self->{match_by_refseq}; 
```

Comment:
It would be helpful to state the delimter here; it's unusual to concatenate values on one key, though I see why you've done it here